cfme/tests/control/test_compliance.py

```diff
@@ -3,38 +3,35 @@
 import fauxfactory
 import pytest
 
-from mgmtsystem import exceptions
-
 from cfme.common.vm import VM
 from cfme.control.explorer.policies import VMCompliancePolicy, HostCompliancePolicy
 from cfme.control.explorer.conditions import VMCondition
 from cfme.control.explorer.policy_profiles import PolicyProfile
-from cfme.infrastructure.provider import InfraProvider
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
 from cfme.configure.configuration import AnalysisProfile
 from cfme.web_ui import flash, toolbar
-from fixtures.pytest_store import store
-from utils import testgen, version
-from utils.appliance import Appliance, ApplianceException, provision_appliance
+from cfme import test_requirements
+from utils import testgen, version, conf
+from utils.hosts import setup_providers_hosts_credentials
 from utils.log import logger
 from utils.update import update
 from utils.wait import wait_for
-from urlparse import urlparse
-from cfme import test_requirements
 
-PREFIX = "test_compliance_"
+vddk_url_map = {
+    "5.5": conf.cfme_data.get("basic_info", {}).get("vddk_url").get("v5_5"),
+    "6": conf.cfme_data.get("basic_info", {}).get("vddk_url").get("v6_0"),
+    "6.5": conf.cfme_data.get("basic_info", {}).get("vddk_url").get("v6_5")
+}
 
 pytestmark = [
-    # TODO: Problems with fleecing configuration - revisit later
     pytest.mark.ignore_stream("upstream"),
     pytest.mark.meta(server_roles=["+automate", "+smartstate", "+smartproxy"]),
-    pytest.mark.uncollectif(lambda provider: provider.type in {"scvmm"}),
     pytest.mark.tier(3),
     test_requirements.control
 ]
 
 
-pytest_generate_tests = testgen.generate(
-    [InfraProvider], required_fields=["vm_analysis"], scope="module")
+pytest_generate_tests = testgen.generate([VMwareProvider], scope="module")
```

Comment:
The reason of that change is the absence cfme templates in nested providers. These tests perform smart state analysis which require some preparations. For VMware providers vddk installed on the provider will be enough. RHEV requires direct lun disk mounted and the appliance also must reside on RHEV.